docs/verify_cstor_bds.sh

```diff
@@ -0,0 +1,43 @@
+set -e
+ns=$1
+if [[ $ns == "" ]]; then
+    ns="openebs"
+fi 
+cspList=$(kubectl get csp -o jsonpath='{.items[*].metadata.name}')
+bdList=$(kubectl -n $ns get blockdevices -o jsonpath='{.items[*].metadata.name}')
+
+for csp in $cspList
+do
+        echo "Verifying blockdevices on $csp"
+        pod=$(kubectl -n $ns get pods -l openebs.io/cstor-pool=$csp -o jsonpath="{.items[?(@.status.phase=='Running')].metadata.name}")
+        devlinks=$(kubectl -n $ns exec -it $pod -c cstor-pool -- zpool status -P | grep dev | awk '{print $1}')
+        for devlink in $devlinks
+        do
+                oldbd=""
+                newbd=""
+                for bd in $bdList
+                do
+                        links=$(kubectl -n $ns get blockdevices $bd -o jsonpath="{.spec.devlinks[?(@.kind=='by-id')].links}")
+                        # remove the list [] from by-id/by-path links output
+                        links=$(echo $links | tr -d [ | tr -d ])
+                        for link in $links
+                        do
+                                if [[ "$devlink" == *"$link"* ]]; then
```

Comment:
As discussed this logic will not help in complex cases like user having multiple raidgroups in CSP.